### PR TITLE
[Extra/rest] DB, emitter, flusher for tss, bandtss module

### PR DESCRIPTION
**Pull request by nkitlabs**
Fixed: DB, emitter, flusher for  tss, bandtss module

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by RogerKSI — CHANGES_REQUESTED** [review]

File: flusher/flusher/db.py [hunk 1]
```diff
@@ -394,30 +463,44 @@ def Column(*args, **kwargs):
 data_source_requests = sa.Table(
     "data_source_requests",
     metadata,
-    Column("data_source_id", sa.Integer, sa.ForeignKey("data_sources.id"), primary_key=True),
+    Column(
+        "data_source_id", sa.Integer, sa.ForeignKey("data_sources.id"), primary_key=True
+    ),
     Column("count", sa.Integer),
```
Comment (satawatnack) [inline flusher/flusher/db.py:469]: why the fomat change a lot which one is correct

File: hooks/emitter/bandtss.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+func (h *Hook) handleBandtssEventGroupTransition(ctx sdk.Context, eventIdx int, querier *EventQuerier) {
+	// if transition not found, skip the process. There is a case that the transition message is signed
+	// at the same block as the transition can be executed. The transition status will be updated via
+	// another event (group_transition_success, group_transition_failed).
+	transition, found := h.bandtssKeeper.GetGroupTransition(ctx)
+	if !found {
+		return
+	}
+
+	// set new bandtss members; better approach would be emitting addMember event.
```
Comment (satawatnack) [inline hooks/emitter/bandtss.go:144]: ambiguous comment

File: hooks/emitter/bandtss.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+	signingID := tss.SigningID(common.Atoi(signingIDs[0]))
+	currentGroupID := tss.GroupID(common.Atoi(currentGroupIDs[0]))
+	incomingGroupID := tss.GroupID(common.Atoi(incomingGroupIDs[0]))
+
+	h.emitUpdateBandtssGroupTransitionStatusSuccess(types.GroupTransition{
+		SigningID:       signingID,
+		CurrentGroupID:  currentGroupID,
+		IncomingGroupID: incomingGroupID,
+	})
```
Comment (satawatnack) [inline hooks/emitter/bandtss.go:188]: reorder the params which one should be coming frist currentGroupID or incomingGroupID

File: hooks/emitter/bandtss.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+// If the event is triggered by the proposal, the active proposal event should be emitted next to the event.
+func getCurrentProposalID(eventIdx int, querier *EventQuerier) (id uint64, found bool) {
+	proposalIDEvent, found := querier.FindEventWithTypeAfterIdx(govtypes.EventTypeActiveProposal, eventIdx)
+	if !found {
+		return 0, false
+	}
+
+	for _, attr := range proposalIDEvent.Attributes {
+		if attr.Key == govtypes.AttributeKeyProposalID {
+			return uint64(common.Atoi(attr.Value)), true
```
Comment (satawatnack) [inline hooks/emitter/bandtss.go:227]: Atoui

File: hooks/emitter/bandtss.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+		IncomingGroupID: tss.GroupID(common.Atoi(incomingGroupIDs[0])),
+	})
+}
+
+// handleBandtssEventSigningRequestCreated implements emitter handler for MsgRequestSignature of bandtss.
+func (h *Hook) handleBandtssEventSigningRequestCreated(ctx sdk.Context, evMap common.EvMap) {
+	signingIDs := evMap[types.EventTypeSigningRequestCreated+"."+types.AttributeKeySigningID]
+
+	for _, sid := range signingIDs {
+		signing := h.bandtssKeeper.MustGetSigning(ctx, types.SigningID(uint64(common.Atoi(sid))))
```
Comment (satawatnack) [inline hooks/emitter/bandtss.go:212]: why not use Atoui

File: hooks/emitter/event_querier.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	events []abci.Event
+}
+
+// NewEventQuerier creates a new EventQuerier with the given events.
+func NewEventQuerier(events []abci.Event) *EventQuerier {
+	return &EventQuerier{events: events}
+}
+
+// FindEventWithTypeBeforeIdx returns the last event of the given type that is emitted before the given index.
+func (es *EventQuerier) FindEventWithTypeBeforeIdx(eventType string, idx int) (abci.Event, bool) {
```
Comment (satawatnack) [inline hooks/emitter/event_querier.go:16]: TBD: should we limit this or we can search until zero

**Review by satawatnack — CHANGES_REQUESTED** [review]

File: hooks/emitter/event_querier.go (hunk 6 above)
Comment (nkitlabs) [reply hooks/emitter/event_querier.go:16]: IMO, I think it serves the purpose of this function; cannot know where the matched event type is unless we go through the list.

Comment (satawatnack) [reply hooks/emitter/event_querier.go:16]: got it!

File: flusher/flusher/db.py (hunk 1 above)
Comment (nkitlabs) [reply flusher/flusher/db.py]: fixed

**Review by colmazia — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**